src/data/automation_i18n.ts

```diff
@@ -493,7 +493,21 @@ export const describeCondition = (
     }
 
     if (condition.before) {
-      base += ` before ${condition.before}`;
+      let duration = "";
+
+      if (condition.before_offset) {
+        if (typeof condition.before_offset === "number") {
+          duration = ` offset by ${secondsToDuration(
+            condition.before_offset
+          )!}`;
+        } else if (typeof condition.before_offset === "string") {
+          duration = ` offset by ${condition.before_offset}`;
+        } else {
+          duration = ` offset by ${JSON.stringify(condition.before_offset)}`;
```

Comment:
`src/common/datetime/format_duration.ts`
```suggestion
          duration = ` offset by ${formatDuration(condition.before_offset)}`;
```

Reply:
If accepted, the same change should be applied to line 488 (in the after_offset block)